### ISSUE TITLE
this.auth.getUniverseDomain is not a function

### ISSUE THREAD

**Issue by RoopeHakulinen** [googleapis/nodejs-pubsub#1884]
1) Is this a client library issue or a product issue?
Product

2) Did someone already solve this?
Not that I know of.

3) Do you have a support contract?
No

#### Environment details

  - OS: Ubuntu, Windows, Mac
  - Node.js version: 20.10.0
  - npm version: 10.2.5
  - `@google-cloud/pubsub` version: 4.3.0

#### Steps to reproduce
After upgrading to 4.3.0, there's an error without proper stacktrace due to asynchronous usage that seems to be coming out of the  `createSubscription` method invocation:
```
TypeError: this.auth.getUniverseDomain is not a function
    at GrpcClient.createStub (/usr/engineering/backend/node_modules/google-gax/src/grpc.ts:418:48)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Everything works when reverting to 4.2.0.

**Comment by IchordeDionysos** [comment]
@RoopeHakulinen this helped for me:
https://github.com/googleapis/google-auth-library-nodejs/issues/1702#issuecomment-1934514234

**Comment by feywind** [comment]
@RoopeHakulinen Thanks for the issue. This is a known thing now, just some minimum versions that need to be coordinated. The `npm upgrade` fix will probably work, though it will force any package versions to newest. A more proper fix is coming.

**Comment by feywind** [comment]
tl;dr for the other issue, still seems to be a problem on 4.3.1.

**Comment by Celinesco** [comment]
I know this issue is marked as closed but I'm getting this :
```
{
  "errorType": "TypeError",
  "errorMessage": "this.auth.getUniverseDomain is not a function",
  "trace": [
    "TypeError: this.auth.getUniverseDomain is not a function",
    "    at GrpcClient.createStub (/var/task/node_modules/google-gax/build/src/grpc.js:312:54)"
  ]
}
```
These are my versions:
 google-gax@4.3.8
  └── google-auth-library@9.13.0
  
  I only installed the @google-cloud/resource-manager package
